### PR TITLE
Refactor TimerAction to allow RosTimer to extend

### PR DESCRIPTION
**Pull request by rebecca-butler**
These changes are necessary for ros2/launch_ros#248

Signed-off-by: Rebecca Butler <rebecca@openrobotics.org>

### CODE REVIEW

File: launch/launch/actions/timer_action.py [hunk 1]
```diff
@@ -78,22 +78,22 @@ def __init__(
...
+        self.completed_future: Optional[asyncio.Future] = None
         self.__canceled = False
-        self.__canceled_future = None  # type: Optional[asyncio.Future]
+        self.canceled_future: Optional[asyncio.Future] = None
         self.__cancel_on_shutdown = type_utils.normalize_typed_substitution(
             cancel_on_shutdown, bool)
         self.__logger = launch.logging.get_logger(__name__)
 
-    async def __wait_to_fire_event(self, context):
+    async def wait_to_fire_event(self, context):
```
Comment (jacobperron) [inline launch/launch/actions/timer_action.py:89]: Others might have a different opinion, but I think we should keep a single underscore prefix for this method name (as well as `completed_future` and `canceled_future`) to indicate that this is *not* intended to be user-facing API.

**Review by jacobperron — COMMENTED** [review]
One comment, otherwise LGTM!

File: launch/launch/actions/timer_action.py (hunk 1 above)
Comment (rebecca-butler) [reply launch/launch/actions/timer_action.py]: Makes sense to me, I made the change.

**Comment by rebecca-butler** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14767/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9516/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12433/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14914/)

**Review by jacobperron — APPROVED** [review]

**Comment by jacobperron** [comment]
It might be good to rerun CI to include `test_launch_ros` in the build since that's where the new ROS timer tests are.

**Comment by rebecca-butler** [comment]
> It might be good to rerun CI to include `test_launch_ros` in the build since that's where the new ROS timer tests are.

Good point, I just ran it again with `test_launch_ros` included.

**Pull request merged**